connection.go

```diff
@@ -0,0 +1,420 @@
+package memphis_kafka
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nkeys"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+const (
+	clientReconnectionUpdateSubject = "memphis.clientReconnectionUpdate"
+	clientTypeUpdateSubject         = "memphis.clientTypeUpdate"
+	clientRegisterSubject           = "memphis.registerClient"
+	memphisLearningSubject          = "memphis.schema.learnSchema.%v"
+	memphisRegisterSchemaSubject    = "memphis.tasks.schema.registerSchema.%v"
+	memphisClientUpdatesSubject     = "memphis.updates.%v"
+	memphisGetSchemaSubject         = "memphis.schema.getSchema.%v"
+	memphisErrorSubject             = "memphis.clientErrors"
+)
+
+type Option func(*Options) error
+
+type Options struct {
+	Host string
+}
+
+type RegisterResp struct {
+	ClientID       int    `json:"clientId"`
+	AccountName    string `json:"accountName"`
+	LearningFactor int    `json:"learningFactor"`
+}
+
+type RegisterReq struct {
+	NatsConnectionID string `json:"natsConnectiontId"`
+	Language         string `json:"language"`
+	Version          string `json:"version"`
+}
+
+type ClientReconnectionUpdateReq struct {
+	NewNatsConnectionID string `json:"newNatsConnectiontId"`
+	ClientID            int    `json:"clientId"`
+}
+
+type ClientTypeUpdateReq struct {
+	ClientID int    `json:"clientId"`
+	Type     string `json:"type"`
+}
+
+type ClientUpdateSub struct {
+	ClientID     int
+	Subscription *nats.Subscription
+	UpdateCahn   chan Update
+}
+
+type Update struct {
+	Type    string
+	Payload []byte
+}
+
+type SchemaUpdateReq struct {
+	MsgName  string
+	SchemaID int
+	Desc     []byte
+}
+
+type Client struct {
+	ClientID              int
+	AccountName           string
+	NatsConnectionID      string
+	IsConsumer            bool
+	IsProducer            bool
+	LearningFactor        int
+	LearningFactorCounter int
+	LearningRequestSent   bool
+	GetSchemaRequestSent  bool
+	BrokerConnection      *nats.Conn
+	JSContext             nats.JetStreamContext
+	ProducerProtoDesc     protoreflect.MessageDescriptor
+	ProducerSchemaID      int
+	ConsumerProtoDescMap  map[int]protoreflect.MessageDescriptor
+	ErrorsMsgChan         chan string
+}
+
+var ClientConnection *Client
+
+func Init(token string, config interface{}, options ...Option) {
+	opts := GetDefaultOptions()
+	for _, opt := range options {
+		if opt != nil {
+			if err := opt(&opts); err != nil {
+				fmt.Println("memphis: error initializing memphis: Wrong option")
+				return
+			}
+		}
+	}
+
+	ClientConnection = &Client{}
+
+	err := ClientConnection.InitializeNatsConnection(token, opts.Host)
+	if err != nil {
+		fmt.Println("memphis: error initializing memphis")
```

Comment:
you are not printing the error itself - now pay attention to that, print it, and make sure it looks good and not something like
memphis: some error... : memphis: some error